lms/djangoapps/courseware/module_render.py

```diff
@@ -920,39 +925,31 @@ def handle_xblock_callback_noauth(request, course_id, usage_id, handler, suffix=
         return _invoke_xblock_handler(request, course_id, usage_id, handler, suffix, course=course)
 
 
-def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
+class XblockCallbackView(APIView):
     """
-    Generic view for extensions. This is where AJAX calls go.
-
-    Arguments:
-        request (Request): Django request.
-        course_id (str): Course containing the block
-        usage_id (str)
-        handler (str)
-        suffix (str)
-
-    Raises:
-        Http404: If the course is not found in the modulestore.
+    Class-based view for extensions. This is where AJAX calls go.
+    Return 403 error if the user is not logged in. Raises Http404 if
+    the location and course_id do not identify a valid module, the module is
+    not accessible by the user, or the module raises NotFoundError. If the
+    module raises any other error, it will escape this function.
     """
-    # NOTE (CCB): Allow anonymous GET calls (e.g. for transcripts). Modifying this view is simpler than updating
```

Comment:
Anonymous access is still needed.

Reply:
Understood. It has been moved to `_get_course_and_invoke_handler` since all the logic was ported over to that method. Its here now: https://github.com/edx/edx-platform/pull/15940/files/60af85cba30ca69d821eb0a3c485cc8dbbc8b96a#diff-b365e6e4f2d48b57c4e934dbd230e3e4R1094

Reply:
Are you a thumbsup on this @clintonb?